www/src/Addons/Wii.tsx

```diff
@@ -43,91 +47,280 @@ export const wiiState = {
+    { id: 'Turntable', value: 5, inputs: { analog: [ { id: 'Stick', axes: [ { id: 'X', axis: 0 }, { id: 'Y', axis: 1 } ] }, { id: 'LeftTurntable', axes: [ { id: 'Trigger', axis: 0 } ] }, { id: 'RightTurntable', axes: [ { id: 'Trigger', axis: 2 } ] }, { id: 'Fader', axes: [ { id: 'Trigger', axis: 4 } ] }, { id: 'Effects', axes: [ { id: 'Trigger', axis: 5 } ] } ], digital: [ { id: 'LeftGreen' }, { id: 'LeftRed' }, { id: 'LeftBlue' }, { id: 'RightGreen' }, { id: 'RightRed' }, { id: 'RightBlue' }, { id: 'Euphoria' }, { id: 'Minus' }, { id: 'Plus' } ] } },
+    { id: 'Taiko',     value: 6, inputs: { analog: [], digital: [ { id: 'DonLeft' }, { id: 'KatLeft' }, { id: 'DonRight' }, { id: 'KatRight' } ] } },
+];
+
+const WII_JOYSTICK_MODES = [
+	{ label: 'None', value: 0, options: {'x': 0, 'y': 0} },
+	{ label: 'Left Analog', value: 1, options: {'x': 1, 'y': 2} },
+	{ label: 'Right Analog', value: 2, options: {'x': 3, 'y': 4} },
+	{ label: 'D-Pad', value: 3, options: {'x': 5, 'y': 6} },
+];
+
 const Wii = ({ values, errors, handleChange, handleCheckbox }) => {
 	const { t } = useTranslation();
+    const [wiiControls, setWiiControls] = useState(baseWiiControls);
+    const [selectedControls] = useState(baseWiiControls);
+    const { setLoading } = useContext(AppContext);
+
+	useEffect(() => {
+		async function fetchData() {
+            let wc = await WebApi.getWiiControls(setLoading);
+			setWiiControls(wc);
+		}
+
+		fetchData();
+	}, [setWiiControls, selectedControls]);
+
+    const ANALOG_SINGLE_AXIS_MODES = [
+        { label: 'None',                 value: 0 },
+        { label: 'Left Trigger',         value: 7 },
+        { label: 'Right Trigger',        value: 8 },
+        { label: 'Left Analog X+ Axis',  value: 9 },
+        { label: 'Left Analog X- Axis',  value: 10 },
+        { label: 'Left Analog Y+ Axis',  value: 11 },
+        { label: 'Left Analog Y- Axis',  value: 12 },
+        { label: 'Right Analog X+ Axis', value: 13 },
+        { label: 'Right Analog X- Axis', value: 14 },
+        { label: 'Right Analog Y+ Axis', value: 15 },
+        { label: 'Right Analog Y- Axis', value: 16 },
+    ];
+    
+    const saveWiiOptions = async () => {
+        let wiiButtonsMap = Array.from(document.querySelectorAll('#WiiExtensionAddonOptions .wii-buttons'));
+        let wiiMap = wiiButtonsMap
+            .reduce(
+                (o, i) => Object.assign(o, { [i.getAttribute('controlid')+'.button'+i.getAttribute('buttonid')]: parseInt(i.value) }), {});
+
+        let wiiAnalogMap = Array.from(document.querySelectorAll('#WiiExtensionAddonOptions .wii-analogs'));
+        wiiMap = wiiAnalogMap
+        .reduce(
+            (o, i) => {
+                let modeID = i.value;
+                let joyMode = WII_JOYSTICK_MODES.filter((o,i) => o.value == modeID)[0];
```

Comment:
Filter is usually used for retrieving one or more based on the predicate.
For one item `Array.find` is usually the way.
Since you only need one here we can simplify it a tiny bit and make it reusable
```javascript
const getJoystickMode = (searchValue: number) =>
	WII_JOYSTICK_MODES.find(({ value }) => value === searchValue);
```

Reply:
This worked nicely with some adjustments for number type.